R/meta.R

```diff
@@ -0,0 +1,52 @@
+
+.roxygen_meta <- new.env(parent = emptyenv())
+
+roxy_meta_get <- function(name, default = NULL) {
+  if (exists(name, envir = .roxygen_meta))
+    get(name, envir = .roxygen_meta)
+  else
+    NULL
+}
+
+roxy_meta_set <- function(key, value) {
+  assign(key, value, envir = .roxygen_meta)
+}
+
+roxy_meta_load <- function(base_path = getwd()) {
+
+  meta_dir <- file.path(base_path, "man/roxygen")
+  if (!utils::file_test("-d", meta_dir))
+    return(FALSE)
+
+  pattern <- "^meta[.][Rr]"
+  meta_files <- list.files(meta_dir, pattern = pattern, full.names = TRUE)
+  if (length(meta_files) == 0)
+    return(FALSE)
+
+  if (length(meta_files) > 1) {
+    rlang::abort("Multiple 'man/roxygen/meta.R' files found")
+  }
+
+  parsed <- tryCatch(parse(meta_files), error = identity)
```

Comment:
Oh just noticed that you're checking for the result of `tryCatch()`; I think that's generally bad practice and instead you should do:

```R
parsed <- tryCatch(
  parse(meta_files),
  error = function(cnd) {
     message <- "Could not parse 'man/roxygen/meta.R'"
     rlang::abort(message, parent = cnd)
  }
}